### PR TITLE
A better Clippy integration

### PR DESCRIPTION
**Pull request by KSXGitHub**
Fix https://github.com/svenstaro/miniserve/issues/93
Close https://github.com/svenstaro/miniserve/pull/94

## Features

* ~~Clippy on Stable Rust that does not allow failures.~~
* ~~Clippy on Nightly Rust that allow failures.~~
* Clippy on pinned nightly.

### CODE REVIEW

File: rls.toml [hunk 1]
```diff
@@ -0,0 +1 @@
+clippy_preference = "on"
```
Comment (svenstaro) [inline rls.toml]: Is this file actually currently being read by anything? Googling for it, I find some bits of old information but that doesn't seem current.

Comment (KSXGitHub) [reply rls.toml]: Hmm, no wonder it had no effects on my editor. What's the new file?

Comment (svenstaro) [reply rls.toml]: I think it depends on the editor. In vim, it even depends on the particular LSP plugin. In VScode, you apparently set it in settings.json.

Comment (KSXGitHub) [reply rls.toml]: I'll just remove this then.

**Comment by KSXGitHub** [comment]
Here's the thing: "#[feature(...)]` is not allowed in stable channel". This makes it impossible to use Clippy on Stable Rust. It seems that the only option left is to pin a Nightly version.

**Comment by svenstaro** [comment]
Alright, well let's just do that then and see how it goes.

**Comment by svenstaro** [comment]
Cool!, let's do it like this. Good work.

**Pull request merged**